solidity/contracts/PhasedEscrow.sol

```diff
@@ -70,26 +70,127 @@ contract PhasedEscrow is Ownable {
     }
 }
 
-interface ICurveRewards {
-    function notifyRewardAmount(uint256 amount) external;
+/// @title BatchedPhasedEscrow
+/// @notice A token holder contract allowing contract owner to approve a set of
+///         beneficiaries of tokens held by the contract, to appoint a separate
+///         drawee role, and allowing that drawee to withdraw tokens to approved
+///         beneficiaries in phases.
+contract BatchedPhasedEscrow is Ownable {
```

Comment:
As we don't extend `PhasedEscrow` maybe we don't need to mention `Phased` escrow in the name? Would `BatchedEscrow` be fine?

Also maybe this contract should be placed in a separate file as we're not referring to `PhasedEscrow`?

Reply:
This contract is still a _phased_ escrow though. The reason why we introduced `PhasedEscrow` was that the standard `Escrow` allows transferring funds only in one shot. `PhasedEscrow` allows to transfer first X amount, then Y amount, then Z amount - that's the main distinction. And this contract is still a *phased* escrow.

Reply:
It's confusing as this contract doesn't extend `PhasedEscrow`.

Reply:
Depends on how you look at it - if you look from a perspective of inheritance then yes, I do agree if it's confusing. If you look from a perspective what those contracts offer, then I think it's not. `BatchedPhasedEscrow` explains what it's doing - allows withdrawing tokens in batches, and in phases. `PhasedEscrow` - allows withdrawing tokens in phases. If we named it just `BatchedEscrow`, the name would not reflect the full functionality.

Reply:
I don't fully agree, but if `BatchedPhasedEscrow` won't have a relation to `PhasedEscrow` we should move it to a separate file to reduce confusion. Perfectly to a dedicated directory where all escrow and beneficiary related stuff will be placed.

Reply:
Can we do it in a separate PR? I started pulling it ou but that's a lot of work.

Reply:
Ok, let's create an issue or PR for that not to loose it.

Reply:
I have already started the work locally on it, I should open PR tomorrow morning, no issue needed.